pkg/operator/cluster/ceph/osd/pod.go

```diff
@@ -40,70 +44,152 @@ const (
 	locationEnvVarName        = "ROOK_LOCATION"
 )
 
-func (c *Cluster) makeDaemonSet(selection rookalpha.Selection, config rookalpha.Config) *extensions.DaemonSet {
-	podSpec := c.podTemplateSpec(nil, selection, c.resources, config)
-	return &extensions.DaemonSet{
+func (c *Cluster) makeJob(nodeName string, devices []rookalpha.Device,
+	selection rookalpha.Selection, resources v1.ResourceRequirements, config rookalpha.Config) *batch.Job {
+
+	podSpec := c.podTemplateSpec(devices, selection, resources, config, true /* prepare-only */, v1.RestartPolicyOnFailure)
+	podSpec.Spec.NodeSelector = map[string]string{apis.LabelHostname: nodeName}
+
+	return &batch.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:            appName,
+			Name:            fmt.Sprintf(prepareAppNameFmt, nodeName),
 			Namespace:       c.Namespace,
 			OwnerReferences: []metav1.OwnerReference{c.ownerRef},
 			Labels: map[string]string{
 				k8sutil.AppAttr:     appName,
 				k8sutil.ClusterAttr: c.Namespace,
 			},
 		},
-		Spec: extensions.DaemonSetSpec{
-			UpdateStrategy: extensions.DaemonSetUpdateStrategy{
-				Type: extensions.RollingUpdateDaemonSetStrategyType,
-			},
+		Spec: batch.JobSpec{
 			Template: podSpec,
 		},
 	}
 }
 
-func (c *Cluster) makeReplicaSet(nodeName string, devices []rookalpha.Device,
-	selection rookalpha.Selection, resources v1.ResourceRequirements, config rookalpha.Config) *extensions.ReplicaSet {
-
-	podSpec := c.podTemplateSpec(devices, selection, resources, config)
-	podSpec.Spec.NodeSelector = map[string]string{apis.LabelHostname: nodeName}
+func (c *Cluster) makeOSDDeployment(nodeName string, devices []rookalpha.Device, selection rookalpha.Selection, resources v1.ResourceRequirements, osd OSDInfo) *extensions.Deployment {
 	replicaCount := int32(1)
+	volumeMounts := []v1.VolumeMount{
+		{Name: k8sutil.DataDirVolume, MountPath: k8sutil.DataDir},
+		k8sutil.ConfigOverrideMount(),
+	}
+	volumes := []v1.Volume{k8sutil.ConfigOverrideVolume()}
+	if c.dataDirHostPath != "" {
+		// the user has specified a host path to use for the data dir, use that instead
+		dataDirSource := v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: c.dataDirHostPath}}
+		volumes = append(volumes, v1.Volume{Name: k8sutil.DataDirVolume, VolumeSource: dataDirSource})
+	}
+
+	// by default, don't define any volume config unless it is required
+	if len(devices) > 0 || selection.MetadataDevice != "" {
+		// create volume config for the data dir and /dev so the pod can access devices on the host
+		devVolume := v1.Volume{Name: "devices", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/dev"}}}
```

Comment:
as @leseb mentioned, can this be a more specific mount to /dev/by-partuuid/<uuid>

Reply:
yes, that's surely the way to go. i would like to have this fix in a followup when we are able to use persistent names instead of device names (sda,etc). Then we can isolate the osd pod to /dev/disk.